### PR TITLE
[JUJU-4400] Rework more mgotests

### PR DESCRIPTION
**Pull request by wallyworld**
More mongo test removal. Most of the LOC churn is moving large chunks of test code.

The httpserver worker is converted from a mongo test to a unit test.

The api package had end to end tests for macaroon login and discharge. But many of these tests were duplicated in the apiserver package. So the tests which did not exist in apiserver were moved across. And a new test file was created to hold just the macaroon tests.

Also rename `State` to `Client` - I think this is one of the last ones with this misnaming.

Driveby - revert to use mgo test package in agentbootstrap tests because the test suite starts mongo manually and using MgoTestpackage makes this easier to discover.

## QA steps

run unit tests

### CODE REVIEW

File: api/agent/agent/client_test.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
+// Copyright 2012, 2013 Canonical Ltd.
```
Comment (wallyworld) [inline api/agent/agent/client_test.go:1]: moved from machine_test.go

File: apiserver/admin_test.go [hunk 2]
```diff
@@ -872,251 +866,6 @@ func (s *loginSuite) TestLoginUpdatesLastLoginAndConnection(c *gc.C) {
 	c.Assert(when, jc.Almost, now)
 }
 
-var _ = gc.Suite(&macaroonLoginSuite{})
```
Comment (wallyworld) [inline apiserver/admin_test.go:875]: moved to macaroon_test.go

**Review by hpidcock — APPROVED** [review]
🍰

File: worker/httpserver/tls.go [hunk 3]
```diff
@@ -31,23 +29,19 @@ func aggregateSNIGetter(getters ...SNIGetterFunc) SNIGetterFunc {
 			}
 		}
 		return nil, lastErr
-	})
+	}
 }
 
 // NewTLSConfig returns the TLS configuration for the HTTP server to use
 // based on controller configuration stored in the state database.
```
Comment (manadart) [inline worker/httpserver/tls.go:36]: I think we need to slow down here and give some attention to comments. This one no longer makes sense, and is now a potential source of confusion for readers.

**Pull request merged**